fury/tests/test_decorators.py

```diff
@@ -49,3 +49,46 @@ def f():
     del HAVE_AMODULE
     f.__doc__ = docstring
     npt.assert_raises(NameError, doctest_skip_parser, f)
+
+
+def test_warn_on_args_to_kwargs():
+    @warn_on_args_to_kwargs()
+    def func(a, b, *, c, d=4, e=5):
+        return a + b + c + d + e
+
+    npt.assert_equal(func(1, 2, c=3, d=4, e=5), 15)
+    npt.assert_equal(func(1, 2, c=3, d=5), 16)
+    npt.assert_equal(func(1, 2, c=3), 15)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 3, 4, 5), 15)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 3, 6), 17)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 10), 22)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 10, d=10, e=19), 42)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 10, e=10), 27)
+    with npt.assert_warns(UserWarning):
+        npt.assert_equal(func(1, 2, 10, d=10), 28)
+    npt.assert_raises(TypeError, func, 1, 2, d=4, e=5)
+    npt.assert_raises(TypeError, func, 1, 2, e=10)
+    npt.assert_raises(TypeError, func, 1, 2, d=10)
+    npt.assert_raises(TypeError, func, 1, 3)
+    npt.assert_raises(TypeError, func, 1)
+
+
+# def test_warn_on_args_to_kwargs_with_versions():
```

Comment:
Will this function stay commented? or is it WIP?

Reply:
Yes it's WIP

Reply:
Will u check different states for different versions as well

Reply:
No, it'll one test for every version.